src/backends/mod.rs

```diff
@@ -13,6 +13,9 @@ pub enum ConsumerError {
     #[error("End of partition reached")]
     EndOfPartition,
 
+    #[error("Invalid consumer state")]
+    InvalidConsumerState,
```

Comment:
I would add some details on what the invalid state was. 
You can make the error message parametric and provide some details 
https://docs.rs/thiserror/latest/thiserror/#example